### PR TITLE
Feature/core 5521 - Added support for following paging during entity retrieval. Some refactoring.

### PR DESCRIPTION
**Pull request by zilladotexe**

### CODE REVIEW

File: src/main/java/com/manywho/services/azure/facades/AzureEntityFetcher.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        List<ODataEntity> entities = new ArrayList<ODataEntity>();
+        
+        ODataEntitySetRequest<ODataEntitySet> entitySetRequest = retrieveRequestFactory.getEntitySetRequest(uri);
+        addAuthHeader(entitySetRequest, token);
+        ODataRetrieveResponse<ODataEntitySet> sitesEntitySetResponse = entitySetRequest.execute();
+        
+        ODataEntitySet body = sitesEntitySetResponse.getBody();
+        entities.addAll(body.getEntities());
+        
+        if(body.getNext() != null){
```
Comment (josecollazzi) [inline src/main/java/com/manywho/services/azure/facades/AzureEntityFetcher.java:50]: @domholmes If the filter by userId is empty. Are those lines fetching all the users in the system?

**Review by josecollazzi — CHANGES_REQUESTED** [review]

File: src/main/java/com/manywho/services/azure/facades/AzureEntityFetcher.java (hunk 1 above)
Comment (zilladotexe) [reply src/main/java/com/manywho/services/azure/facades/AzureEntityFetcher.java:50]: Yes, the filter segment is not added to the url if `filter` is null or empty.

**Comment by zilladotexe** [comment]
We've decided to release this once [FLOW-743] has been completed as both are required for a good user experience, particularly when a large amount of users are returned from the service.

[FLOW-743]: https://boomii.atlassian.net/browse/FLOW-743

**Comment by zilladotexe** [comment]
This work was superceded by a new piece to correct search functionality, negating the need for large amount of entities to be pulled.

**Pull request closed without merging**